tcp/server.go

```diff
@@ -50,40 +50,36 @@ func (s Server) Addr() net.Addr {
 }
 
 // handle handles incoming requests
-func handle(conn net.Conn, c Callback) error {
```

Comment:
I would return an error here still too, you can change the way handle is called. To use an anonymous goroutine, and the error is handled in that, because like this execution isn't ended.